### PR TITLE
Feature#35 generate reports

### LINKED ISSUE #35: Generate reports in index report views
# REQUIREMENTS
Create function to generate reports by day, month and average within the report view.
Reports must be overwritten in the report view according to the button that is pressed.

# Acceptance Criteria
- [ ] Function generate report

### PR DESCRIPTION
**Pull request by RobertoCarMed**
# Description
Views were added for reports by day, month and the average time employees checked in / out.

The corresponding views were also added to the routes.rb file to be able to access them.

resolve #35

### CODE REVIEW

**Review by SEVERO96 — DISMISSED** [review]

File: app/views/reports/index.html.erb [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+      <tr>
+        <th class="col">Id</th>
+        <th class="col">Check In</th>
+        <th class="col">Check Out</th>
+        <th class="col">Employee Id</th>
+        
+      </tr>
+    </thead>
+    <tbody>
+      <% @reports = Report.all %>
```
Comment (jclopezdev) [inline app/views/reports/index.html.erb]: This line shouldn't be here, this must go in the controller

File: app/views/reports/report_average.html.erb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<%= link_to "back", reports_path, class: "btn btn-info" %>
+
+<%
```
Comment (jclopezdev) [inline app/views/reports/report_average.html.erb]: I recommend you to move all this logic to a different place, remember that the Views in MVC are only to show the information.

Check these links and let's talk in Slack if you want
https://stackoverflow.com/a/13179685
https://jtway.co/cleaning-up-your-rails-views-with-view-objects-42cf048ea491

**Review by jclopezdev — CHANGES_REQUESTED** [review]
I left some comments, but if you want we can check it in Slack

File: app/views/reports/report_day.html.erb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        <th class="col">Id</th>
+        <th class="col">Check In</th>
+        <th class="col">Check Out</th>
+        <th class="col">Employee Id</th>
+        
+      </tr>
+    </thead>
+    <tbody>
+      <% 
+        @reports = Report.all
```
Comment (YilKanda) [inline app/views/reports/report_day.html.erb]: This must be in the Controller.

File: app/views/reports/report_month.html.erb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        <th class="col">Id</th>
+        <th class="col">Check In</th>
+        <th class="col">Check Out</th>
+        <th class="col">Employee Id</th>
+        
+      </tr>
+    </thead>
+    <tbody>
+      <% 
+        @reports = Report.all
```
Comment (YilKanda) [inline app/views/reports/report_month.html.erb]: This must be in the Controller.

File: app/views/reports/report_month.html.erb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          </tr>
+        <%else%>
+
+        <%end%>          
+      <% end %>
+      <%if reports_today == 0%>
+        <h1>No reports today</h1>
+      <%end%>
+    </tbody>
+  </table>
```
Comment (YilKanda) [inline app/views/reports/report_month.html.erb:35]: Add a blank line.

File: app/views/reports/report_day.html.erb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          </tr>
+        <%else%>
+
+        <%end%>          
+      <% end %>
+      <%if reports_today == 0%>
+        <h1>No reports today</h1>
+      <%end%>
+    </tbody>
+  </table>
```
Comment (YilKanda) [inline app/views/reports/report_day.html.erb:35]: Add a blank line.

File: app/views/reports/report_average.html.erb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      <h4> <%= "#{average_hour_check_out}:0#{average_minutes_check_out}" %> </h4>
+    <% else %>
+
+      <h4> <%= "#{average_hour_check_out}:#{average_minutes_check_out}" %> </h4>
+    
+    <%end%>
+
+
+  </div>  
+</div>
```
Comment (YilKanda) [inline app/views/reports/report_average.html.erb:14]: Add a blank line.

File: app/views/reports/report_average.html.erb (hunk 2 above)
Comment (YilKanda) [reply app/views/reports/report_average.html.erb]: Fully agree with @jclopezdev

**Review by YilKanda — CHANGES_REQUESTED** [review]
You have code that must go into the controller in the view.

This is a problem, as you are putting sensitive information in a place where it can be easily viewed.

**Review by SEVERO96 — APPROVED** [review]

File: app/controllers/reports_controller.rb [hunk 8]
```diff
@@ -69,4 +71,9 @@ def date_match(employee)
       end
     end
   end
+
+  def get_reports
+    return Report.all
```
Comment (YilKanda) [inline app/controllers/reports_controller.rb:76]: Remember that Ruby by default returns the last value that is placed in the def. So the return is not necessary.

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by xFerchoVJ — APPROVED** [review]
LGTM

File: app/views/reports/report_month.html.erb (hunk 5 above)
Comment (SampabloC) [reply app/views/reports/report_month.html.erb:35]: thanks for your feedback, we are  going to correct that

**Pull request merged**